platforms/software/dependency-management/src/main/java/org/gradle/internal/locking/InvalidLockFileException.java

```diff
@@ -16,12 +16,13 @@
 
 package org.gradle.internal.locking;
 
-import org.gradle.internal.exceptions.ResolutionProvider;
+import org.gradle.api.GradleException;
+import org.gradle.api.ResolutionProvider;
 
 import java.util.Collections;
 import java.util.List;
 
-public class InvalidLockFileException extends RuntimeException implements ResolutionProvider {
+public class InvalidLockFileException extends GradleException implements ResolutionProvider {
```

Comment:
No longer need to `implements ResolutionProvider` since we get that from `GradleException` now. 

We should also pass `resolutionMessage` into `super` instead of implementing `getResolutions` ourselves